### PR TITLE
Add system test configuration using settings file

### PR DESCRIPTION
**Pull request by altenstedt**
This commit will replace the TokenHttpClient class with a custom delegate handler in combination with a base test class.

This simplifies the code and also means that we also where able to implement a testsettings.json file that enables more real-world scenarios such as running system tests on a CI server with configuration.

### CODE REVIEW

File: rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/HealthTests.cs [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+    {
+    }
 
     [Fact]
     public async Task LivenessAnonymous_ShouldReturn200AndCorrectVersion()
     {
-        var client = new HttpClient();
-
-        var response = await client.GetAsync(new Uri(baseUri, "/api/health/live"));
+        var response = await HttpClient.GetAsync("/api/health/live");
```
Comment (TobiasAhnoff) [inline rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/HealthTests.cs]: This should be a call without a token,  var client = new HttpClient();

File: rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/BaseTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        Configuration = builder.Build();
+
+        HttpClient = new HttpClient(new ClientCredentialsDelegatingHandler(Configuration));
+        HttpClient.BaseAddress = new Uri(Configuration["BaseAddress:Uri"]!);
+        
+        output.WriteLine("API base uri: {0}", HttpClient.BaseAddress);
+    }
+
+    protected HttpClient HttpClient { get; }
```
Comment (TobiasAhnoff) [inline rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/BaseTests.cs]: Rename to something taht is not confused with the class HttpClient, e g TokenHttpClient or HttpClientWithToken?

File: rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/HealthTests.cs (hunk 1 above)
Comment (altenstedt) [reply rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/HealthTests.cs]: Fixed in amended commit.

File: rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/BaseTests.cs (hunk 2 above)
Comment (altenstedt) [reply rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/BaseTests.cs]: Fixed in amended commit.

**Review by TobiasAhnoff — APPROVED** [review]

**Pull request merged**